### ISSUE TITLE
[folding] [lua] Weird code folding with Lua

### ISSUE THREAD

**Issue by zethon** [Microsoft/vscode#3602]
A minor observation about an otherwise great feature! Using the Insiders build, the following Lua code causes weird folding:

```
function someFunc()

    local longString = [[this is a long
string that spans multiple lines
and has some weird folding]]

    print longString
end
```

You'll notice that a + shows up on the same "function someFunc()" line _and_ on the last line of the longString declaration. The first + folds up to the declaration line and the second + folds the rest of the function.

**Comment by Tyriar** [comment]
Code folding is currently based on indentation, this is blocked by language-aware folding https://github.com/Microsoft/vscode/issues/3422

**Comment by aeschli** [comment]
duplicate of #3422
